tests/test_api.py

```diff
@@ -29,38 +24,18 @@ def bliss_client(cwd, cfg):
     return client
 
 
-@pytest.fixture(scope="class")
-def cached_data_path_api(bliss_client):
-    bliss_client.cached_data_path = bliss_client.cwd + "/data/cached_dataset"
-    bliss_client.generate(n_batches=3, batch_size=5, max_images_per_file=10)
-    return bliss_client.cached_data_path
-
-
-@pytest.fixture(scope="class")
-def weight_save_path(bliss_client, cfg):
-    """Train model for 1 epoch and return path to saved model."""
-    weight_save_path = "tutorial_encoder/0_fixture.pt"
-    bliss_client.train_on_cached_data(
-        weight_save_path=weight_save_path,
-        train_n_batches=1,
-        batch_size=5,
-        val_split_file_idxs=[1],
-        training={
-            "n_epochs": 1,
-            "trainer": {"check_val_every_n_epoch": 1, "log_every_n_steps": 1},
-            "pretrained_weights": cfg.predict.weight_save_path,
-        },
-    )
-    return weight_save_path
-
-
-@pytest.mark.usefixtures("bliss_client", "cached_data_path_api", "weight_save_path")
+@pytest.mark.usefixtures("bliss_client")
 class TestApi:
-    def test_get_dataset_file(self, bliss_client, cached_data_path_api):
-        bliss_client.cached_data_path = cached_data_path_api
+    def test_get_dataset_file(self, cfg, bliss_client):
+        bliss_client.cached_data_path = cfg.paths.data + "/tests/multiband_data"
         dataset0 = bliss_client.get_dataset_file(filename="dataset_0.pt")
         assert isinstance(dataset0, list), "dataset0 must be a list"
 
+    def test_generate(self, bliss_client, monkeypatch):
+        monkeypatch.setattr(api, "_generate", mock_generate)
+        bliss_client.generate(n_batches=2, batch_size=1, max_images_per_file=2)
+        assert Path(bliss_client.cached_data_path).exists()
```

Comment:
I think we want to at least generate one dataset file and check that it exists in this `bliss_client.cached_data_path`? So maybe we shouldn't mock `_generate`? Or at least return a test file from `mock_generate`?

Reply:
Changed to mock the simulator instead, since we already test simulate. So `test_api.test_generate` will run through all of the generate code and write (empty) files.